frontend/src/components/common/ContextMenu.js

```diff
@@ -0,0 +1,148 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import MetacodeSelect from '../MetacodeSelect'
+
+class ContextMenu extends Component {
+  static propTypes = {
+    topicId: PropTypes.string,
+    contextNode: PropTypes.object,
+    contextEdge: PropTypes.object,
+    contextPos: PropTypes.object,
+    contextFetchingSiblingsData: PropTypes.bool,
+    contextSiblingsData: PropTypes.object,
+    metacodeSets: PropTypes.array,
+    contextReset: PropTypes.func,
+    contextDelete: PropTypes.func,
+    contextRemove: PropTypes.func,
+    contextHide: PropTypes.func,
+    contextCenterOn: PropTypes.func,
+    contextPopoutTopic: PropTypes.func,
+    contextUpdatePermissions: PropTypes.func,
+    contextOnMetacodeSelect: PropTypes.func,
+    contextFetchSiblings: PropTypes.func,
+    contextPopulateSiblings: PropTypes.func
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      populateSiblingsSent: false
+    }
+  }
+
+  render () {
+    const { contextNode, contextPos, contextOnMetacodeSelect, metacodeSets,
+            contextSiblingsData, contextFetchSiblings,
+            contextPopulateSiblings, contextFetchingSiblingsData,
+            topicId } = this.props
+    const style = {
+      position: 'absolute',
+      top: contextPos.y + 'px',
+      left: contextPos.x + 'px'
+    }
+
+    const populateSiblings = () => {
+      if (!this.state.populateSiblingsSent) {
+        contextPopulateSiblings(contextNode.id)
+        this.setState({populateSiblingsSent: true})
+      }
+    }
+
+    // TODO: add checks for logged in and other context
+    return <div className="rightclickmenu" style={style}>
+      <ul>
+        <li className="rc-hide">
+          <div className="rc-icon" />Hide until refresh
+            <div className="rc-keyboard">Ctrl+H</div>
+        </li>
+        <li className="rc-remove ">
+          <div className="rc-icon" />Remove from map
+            <div className="rc-keyboard">Ctrl+M</div>
+        </li>
+        <li className="rc-delete ">
+          <div className="rc-icon" />Delete
+            <div className="rc-keyboard">Ctrl+D</div>
+        </li>
+        <li className="rc-popout">
+          <div className="rc-icon" />Open in new tab
+        </li>
+        <li className="rc-spacer"></li>
```

Comment:
I'd love to see these broken out into helper functions, just to make the render function smalller and easier to parse

Reply:
I do get this is a draft lol but when it's ready for prime time I mean